app-backend/common/src/main/scala/cache/CacheClient.scala

```diff
@@ -0,0 +1,83 @@
+package com.azavea.rf.common.cache
+
+import java.util.concurrent.Executors
+
+import net.spy.memcached._
+
+import scala.concurrent._
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import com.azavea.rf.common.Config
+import cats.data._
+import com.google.common.util.concurrent.ThreadFactoryBuilder
+import com.typesafe.scalalogging.LazyLogging
+
+object CacheClientThreadPool {
+  implicit lazy val ec: ExecutionContext =
+    ExecutionContext.fromExecutor(
+      Executors.newFixedThreadPool(
+        Config.memcached.threads,
+        new ThreadFactoryBuilder().setNameFormat("cache-client-%d").build()
+      )
+    )
+}
+
+class CacheClient(client: => MemcachedClient) extends LazyLogging {
+
+  import CacheClientThreadPool._
+
+  val cacheEnabled = Config.memcached.enabled
+
+  def delete(key: String) =
+    if(cacheEnabled) {
+      client.delete(key)
+    }
+
+  def setValue[T](key: String, value: T, ttlSeconds: Int = 0): Unit = {
+    Future {
+      client.set(key, ttlSeconds, value)
+    }
```

Comment:
We came across the internal [`ExecutorService`](https://github.com/awslabs/aws-elasticache-cluster-client-memcached-for-java/blob/master/src/main/java/net/spy/memcached/DefaultConnectionFactory.java#L324-L352) too. It seems like that's something worth tuning/evaluating. I opened https://github.com/azavea/raster-foundry/issues/2136 to track it.

As far as the code snippets go, we should be referencing [this](https://github.com/awslabs/aws-elasticache-cluster-client-memcached-for-java/blob/master/src/main/java/net/spy/memcached/MemcachedClient.java) repository, right?